src/components/Select/components/Control/Control.tsx

```diff
@@ -24,10 +29,10 @@ type ControlProps = {
     placeholder?: SelectProps['placeholder'];
     active?: boolean;
     disabled?: boolean;
-    listboxRef?: React.RefObject<List<FlattenOption>>;
+    listboxRef?: React.RefObject<List<FlattenOption<T>>>;
 };
 
-export const Control = React.forwardRef<HTMLButtonElement, ControlProps>((props, ref) => {
+const ControlInner = <T extends unknown>(props: ControlProps<T>, ref: ButtonForwardedRef) => {
```

Comment:
I wonder why we need the generic here in the first place? It comes from SelectOptions that inherits HTMLSelectOption. Do we actually need that inheritance?